### DISCUSSION TITLE
why we have so much difference between cpp file and name of function in asm ?

### DISCUSSION THREAD

**Discussion by coco875** [riidefi/mkw#185] [General]
so I try to understand how to pass asm in Cpp but when I see name of function in symbol.txt we have for example Entity_construct but in cpp file I don't find similary name. so why is so different ? and when I see the issue you have the zone and what he made in global but is not implemented and by edit gen_asm.py I see you can create a base of cpp file for decompilation so why you don't add this in the first start ?

**Comment by riptl** [comment]
I'm not sure I understand what you mean. You can use [`verify_symbols.py`](https://github.com/riidefi/mkw/blob/master/mkwutil/tools/verify_symbols.py) to print differences between symbols.txt and the ELF.

**Comment by riptl** [comment]
@riidefi Propose moving this to the discussions tab

**Reply by snailspeed3** [reply]
Converted to a discussion

**Comment by coco875** [comment]
I mean for exemple BoxColUnit.cpp is between 0x80786ED0 and 0x80786F98 so this mean then in file we have this function :
```0x80786ed0 Entity_construct
0x80786ef4 Entity_destroy
0x80786f34 Entity_init
0x80786f6c unk_80786f6c
0x80786f7c unk_80786f7c
0x80786f98 unk_80786f98
```
but in file I don't find this function

**Comment by snailspeed3** [comment]
Keep in mind the game doesn't have an official symbol map--we can't know with complete certainty what any class was actually named. However, as this class also exists in MK7, we can just use that name -- "BoxColUnit". I have the following as
```
0x80786ef4 BoxColUnit::~BoxColUnit((void))
0x80786f34 BoxColUnit::set((const EGG::Vector3f&, u32, void *, float, float))
0x80786f6c BoxColUnit::makeDynamic((void))
0x80786f7c BoxColUnit::resizeSphere((float, float))
```

**Reply by coco875** [reply]
oh ok it is not actualise, is provisor name ok, and I try to make a file cpp pregenerate so I add this line : `1,source/game/kart/KartAction.cpp,0x805672cc,0x8056a264,,,,,,,,,,` (I found this in issue but is just for try and I edit gen_asm to fusion DOLSrcGenerator and RELSrcGenerator because is the same but with some different) and I obtain many error but I suppose then the first error make also all other error
```
   source\game\kart\KartAction.cpp:396: (33067) illegal forward label orrt\KartAction.cpp
   source\game\kart\KartAction.cpp:396: undefined symbol (qr0) in constant
   source\game\kart\KartAction.cpp:396: expression
   source\game\kart\KartAction.cpp:688: (33067) illegal forward label or
   source\game\kart\KartAction.cpp:688: undefined symbol (qr0) in constant
   source\game\kart\KartAction.cpp:688: expression
   source\game\kart\KartAction.cpp:697: (10159) undefined label 'qr0'
   source\game\kart\KartAction.cpp:704: (10130) '}' expected
```

**Reply by coco875** [reply]
if you want to try [here](https://github.com/coco875/mkw) is my fork

**Reply by snailspeed3** [reply]
Adding `#define qr0 0` should fix it. Ideally this would be put in a common header.

**Reply by coco875** [reply]
ok thank, I try and he fix it but he said he was different so the auto generate file is wrong ?

**Reply by coco875** [reply]
the error : `text: at_src=0x000000d4 at_dst=0x000000d4 at_end=0x0037f120 want=0x0037f04c have=0x0037f04c [FAIL]` but it work perfectly when the game run

**Reply by coco875** [reply]
ok I don't know why but you can define one UNKNOWN_FUNCTION because all other take memory adress of the first
